sockeye/model.py

```diff
@@ -189,6 +191,19 @@ def __init__(self,
 
         self.knn : Optional[layers.KNN] = None
 
+    def set_inference_only(self, inference_only: bool):
+        """
+        Turn inference_only/eval mode on or off.
+        """
+        if self.inference_only != inference_only:
+            self.inference_only = inference_only
+            if self.inference_only:
+                self.output_layer = self.output_layer_script_cached
+            else:
+                self.output_layer = self.output_layer_module_cached
+            self.decoder.set_inference_only(self.inference_only)
+            self.train(not inference_only)
```

Comment:
Do we need to set `self.train` here? Could we say that `self.set_inference_only` controls Sockeye-level optimizations (cell state transform and output layer) while `self.train` controls PyTorch-level optimizations (dropout)? This would decouple the two settings to avoid potential unintended side effects when users change one or the other.